it/controllers/incomeSources/add/BusinessAccountingMethodControllerISpec.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.incomeSources.add
+
+import config.featureswitch.IncomeSources
+import forms.utils.SessionKeys.addBusinessAccountingMethod
+import helpers.ComponentSpecBase
+import helpers.servicemocks.IncomeTaxViewChangeStub
+import play.api.http.Status.{BAD_REQUEST, OK, SEE_OTHER}
+import testConstants.BaseIntegrationTestConstants.testMtditid
+import testConstants.IncomeSourceIntegrationTestConstants.noPropertyOrBusinessResponse
+
+class BusinessAccountingMethodControllerISpec extends ComponentSpecBase {
+
```

Comment:
just wondering if we have similar it:test for agent? or this all incorporate in this single test?

Reply:
I noticed that none of the it:tests had separate agent tests so thought they weren't necessary (I also tried writing them myself but couldn't get anywhere). the controller unit tests have agent tests, however

Reply:
Hmm... I must've been looking in the wrong place as I can see them now. Will get on that